### PR TITLE
Clean and improve tests of codes in /R -> 100% coverage

### PR DESCRIPTION
**Pull request by rbloehm**
This is a PR that cleans a duplicate test (for gather) and adds further tests to spread (with fill) and id, bringing code coverage of files in /R to 100%.
As it is my first pull request into a public repo, I am more than open for comments on how to improve my process of generating a good pull request. Thanks!

### CODE REVIEW

File: tests/testthat/test-gather.R [hunk 1]
```diff
@@ -88,24 +88,6 @@ test_that("overwrites existing vars", {
 
 # Column types ------------------------------------------------------------
 
-test_that("can gather all atomic vectors", {
```
Comment (hadley) [inline tests/testthat/test-gather.R]: Why did you remove these tests?

File: tests/testthat/test-spread.R [hunk 2]
```diff
@@ -2,14 +2,14 @@ context("Spread")
 library(dplyr, warn.conflicts = FALSE)
 
 test_that("order doesn't matter", {
-  df1 <- data.frame(x = c("a", "b"), y = 1:2)
-  df2 <- data.frame(x = c("b", "a"), y = 2:1)
+  df1 <- tibble(x = c("a", "b"), y = 1:2)
```
Comment (hadley) [inline tests/testthat/test-spread.R]: Changing from `data.frame()` to `tibble()` subtly changes the test because `tibble()` doesn't turn strings into factors.

File: tests/testthat/test-spread.R [hunk 3]
```diff
@@ -243,3 +239,21 @@ test_that("spread doesn't convert data frames into tibbles", {
   df <- data.frame(x = c("a", "b"), y = 1:2)
   expect_equal(class(spread(df, x, y)), "data.frame")
 })
+
+test_that("spread with fill replaces explicit missing values", {
+  df <- tibble(key = c("a"), value = c(NA))
```
Comment (hadley) [inline tests/testthat/test-spread.R]: Can you please eliminate the needless `c()` here?

File: tests/testthat/test-underscored.R [hunk 4]
```diff
@@ -2,7 +2,7 @@ context("Deprecated SE variants")
 
 test_that("complete_()", {
   df <- tibble(x = 1:2, y = 1:2, z = 3:4)
-  out <- complete_(df, list("x", ~y))
+  out <- complete_(df, list("x", ~ y))
```
Comment (hadley) [inline tests/testthat/test-underscored.R]: Can you please remove the space? See http://style.tidyverse.org/syntax.html#spacing

File: tests/testthat/test-gather.R (hunk 1 above)
Comment (rbloehm) [reply tests/testthat/test-gather.R]: Because the one below is exactly the same, but does even more. It even has the same name "can gather all atomic vectors".

File: tests/testthat/test-spread.R (hunk 2 above)
Comment (rbloehm) [reply tests/testthat/test-spread.R]: good point, added a factor() to all relevant places in test-spread

File: tests/testthat/test-spread.R (hunk 3 above)
Comment (rbloehm) [reply tests/testthat/test-spread.R]: Sure, also did so in 1 or 2 other places

File: tests/testthat/test-underscored.R (hunk 4 above)
Comment (rbloehm) [reply tests/testthat/test-underscored.R]: Done, in 2 places

File: tests/testthat/test-gather.R (hunk 1 above)
Comment (hadley) [reply tests/testthat/test-gather.R]: Oh, got it!

**Comment by hadley** [comment]
Thank you!!
